tests/module_cpp20/CMakeLists.txt

```diff
@@ -2,11 +2,18 @@ cmake_minimum_required(VERSION 3.28)
 
 project(json_test CXX)
 
-add_executable(json_test)
+set(NLOHMANN_JSON_BUILD_MODULES ON CACHE BOOL "Enable nlohmann.json module support")
 
-target_sources(json_test
-    PRIVATE main.cpp
-    PUBLIC FILE_SET cxx_modules TYPE CXX_MODULES FILES json.cpp)
+add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../.. ${CMAKE_CURRENT_BINARY_DIR}/tests)
```

Comment:
I am still struggling with this line. I will check tomorrow what this should be.